core/Plugin/Report.php

```diff
@@ -318,6 +318,26 @@ public function render()
     }
 
     /**
+     *
+     * Processing a uniqueId for each report, can be used by UIs as a key to match a given report
+     * @return string
+     */
+    public function getId()
+    {
+        $params = $this->getParameters();
+
+        $paramsKey = $this->getModule() . '.' . $this->getAction();
+
+        if (!empty($params)) {
```

Comment:
It is somewhat duplicated logic from `ProcessedReport` but didn't want to make the method static etc. Eventually, `ProcessedReport` will directly work on the report classes instead of some arrays and then we have it only in one place.